### DISCUSSION TITLE
Release 1.2.0rc1, for testing only

### DISCUSSION THREAD

**Discussion by ThomasWaldmann** [borgbackup/borg#6254] [General]
borgbackup 1.2.0rc1 release - do not use this on production backup repositories.

Long changelog:
https://github.com/borgbackup/borg/blob/1.2.0rc1/docs/changes.rst#version-120rc1-2022-02-05

If you use pip to install this, use: pip install pkgconfig ; pip install "borgbackup==1.2.0rc1"

Feedback:
- ticket: https://github.com/borgbackup/borg/issues/4360
- discussion: https://github.com/borgbackup/borg/discussions/xxxx

Note: macos binaries will come later.

<hr /><em>This discussion was created from the release <a href='https://github.com/borgbackup/borg/releases/tag/1.2.0rc1'>Release 1.2.0rc1, for testing only</a>.</em>

**Comment by ThomasWaldmann** [comment]
1.2.0rc1 is released - please test!

1.2.0 final will have the same bugs as rc1 except if you find and report them in the next 2 weeks. :-)

Upgrade notes: https://github.com/borgbackup/borg/pull/6257/files

**Comment by lfam** [comment]
I did some tests, building and deploying 1.2.0rc1 in Guix.

I tested with client at 1.2.0rc1 while the server was 1.1.17, vice versa, and both ends upgraded to 1.2.0rc1. I tested basic functions with a small dataset: creating multiple archives, listing them, extracting them, and mounting them with FUSE. All seems fine.

We are using Python 3.9.9, Cython 0.29.24 (we rebuild the cython files), msgpack 1.0.2, llfuse 1.4.1. We replace the bundled compressors with these: lz4 1.9.3, xxhash 0.8.1, and zstd 1.5.0.

**Comment by ThomasWaldmann** [comment]
@lfam thanks for the feedback, sounds good!

About replacing bundled compressors: if you mean that you changed the source tree of borg and upgraded the bundled source code we have there: that's not necessary. if you use pkgconfig, borg will just find your system-installed libs and headers and use them automagically instead of compiling the bundled compressor code (see top of setup.py)

**Comment by lfam** [comment]
I don't mean that we upgrade the bundled compressors. I mean that we don't use them and instead use the system-installed libraries for them.

**Comment by remyabel2** [comment]
Works fine.

```
$ virtualenv ~/.borg-120
$ source ~/.borg-120/bin/activate.fish
$ pip install pkgconfig
$ pip install "borgbackup==1.2.0rc1"
$ borg init -e repokey-blake2 -p --make-parent-dirs $HOME/second_drive/borg-120-test
# snip
$ borg create -s -p --patterns-from $HOME/backup-secrets-repo/patterns.lst $HOME/second_drive/borg-120-test/::Test $HOME
# snip
Repository: /home/remyabel/second_drive/borg-120-test
Archive name: Test
Archive fingerprint: 39f38407240b212945d659cda964b10a489c0252f43bfd6ecda35635c825ec72
Time (start): Sun, 2022-02-13 22:40:31
Time (end):   Sun, 2022-02-13 23:01:03
Duration: 20 minutes 31.88 seconds
Number of files: 294228
Utilization of max. archive size: 0%
# snip
$ borg list $HOME/second_drive/borg-120-test/
Test                                 Sun, 2022-02-13 22:40:31 [39f38407240b212945d659cda964b10a489c0252f43bfd6ecda35635c825ec72]
$ borg info $HOME/second_drive/borg-120-test/
Repository ID: 0fc210bfee0696b0b1e23b975fc88f5ddf14603d2d2459a8b9f65ba6a7d0b87a
Location: /home/remyabel/second_drive/borg-120-test
Encrypted: Yes (repokey BLAKE2b)
Cache: /home/remyabel/.cache/borg/0fc210bfee0696b0b1e23b975fc88f5ddf14603d2d2459a8b9f65ba6a7d0b87a
Security dir: /home/remyabel/.config/borg/security/0fc210bfee0696b0b1e23b975fc88f5ddf14603d2d2459a8b9f65ba6a7d0b87a
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
All archives:               85.08 GB             70.71 GB             66.70 GB

                       Unique chunks         Total chunks
Chunk index:                  243044               321945
$ borg delete --stats $HOME/second_drive/borg-120-test
```
